target/scripts/startup/setup-stack.sh

```diff
@@ -599,18 +632,36 @@ function _setup_SRS
   postconf 'recipient_canonical_classes = envelope_recipient,header_recipient'
 }
 
-function _setup_dkim
+function _setup_dkim_dmarc
 {
+  if [[ ${ENABLE_OPENDMARC} -eq 1 ]]
+  then
+    _log 'trace' "Adding OpenDMARC to Postfix's milters"
+
+    # shellcheck disable=SC2016
+    sed -i -E 's|^(smtpd_milters =.*)|\1 \$dmarc_milter|g' /etc/postfix/main.cf
```

Comment:
`smtpd_milters = ` expects a comma separated list. This `sed` statement may not be a problem, but the ones below are.

Reply:
Good catch, so our tests are lacking an integration case that should have failed with Postfix being tested for DKIM and DMARC support working all together?

Reply:
Where'd you get this? Postfix docs: "A list of Milter (mail filter) applications for new mail that arrives via the Postfix smtpd(8) server. Specify space or comma as separator. See the MILTER_README document for details."

Reply:
Probably yes. Not sure if a space separated list is somehow valid, however all other options use commas and we should keep it uniformly.

Reply:
Doing it this way is really easy because the `sed` command is easy; using command, you have to handle way more logic about which one is last, which is first, and where to place the comma. So; a space-separated list is easier to maintains. Postfix does not care either way..

Reply:
This should handle this:

```bash
function _postfix_maincf_extend_option() {
	local option value

	_adjust_mtime_for_postfix_maincf
	# get current option and value(s)
	option=$(postconf "${1}")
	# extract current value(s)
	value=$(cut -d' ' -f3- <<< "$option")

	# extend value(s)
	if [ -z "${value}" ]; then
		postconf "${option}${2}"
	else
		postconf "${option}, ${2}"
	fi
}
```

Reply:
Minor nitpick if adding that function, `append` is probably better than `extend` :sweat_smile: 

Doesn't that make removal more fussy too though? I can't recall exactly without looking at the alias utilities, but I know this support was problematic for one format there, possibly it was without `,`.

As long as it's working as-is, I'm fine with the current `sed` approach. I'd sooner rather we delegate more config juggling to something not relying on shell scripts / commands since it's way easier to deal with array like data in a proper language :sweat_smile:

Reply:
> Moreover, a quick look in our Postfix `main.cf` shows that we in fact do not use commas everywhere. I see no issue here.

You are right! I just had a quick look at main.cf and missed that. I was confused because comma was used earlier [here](https://github.com/docker-mailserver/docker-mailserver/blob/cb8e336d256136b79cf58d6becd596cdb56fdb47/target/postfix/main.cf#L96).

Reply:
```
smtpd_milters (default: empty)

    A list of Milter (mail filter) applications for
    new mail that arrives via the Postfix smtpd server.
    Specify space or comma as separator.
```

🎉

Reply:
Something was pretty weird here. I just refreshed the page and read this thread again. When I was answering, I did not see any of Georgs replys. That's why my answers seem to be unsuited 😖 Sorry for that. Also my last post.. just saw that Georg already posted that 🤦

Reply:
No worries :) I actually have no objections to making it uniform, just not in this PR :D

GitHub status is actually also showing degraded performance for Action, Codespaces & API requests, so something is going on..

I will resolve this thread though, because we resolved the matter.